src/api/instance/localInstanceImpl.js

```diff
@@ -0,0 +1,184 @@
+      await this.loadPluginsBudget(),
+    ]);
+    return {
+      pluginGraphs,
+      ledger,
+      weightOverrides,
+      dependencies,
+      pluginsBudget,
+    };
+  }
+
+  async loadCredGraph(): Promise<CredGraph> {
+    const credGraphPath = pathJoin(this._baseDirectory, ...CREDGRAPH_PATH);
+    return await loadJson(credGraphPath, credGraphParser);
+  }
+
+  async writeLedger(ledger: Ledger): Promise<void> {
+    const ledgerPath = pathJoin(this._baseDirectory, ...LEDGER_PATH);
+    return fs.writeFile(ledgerPath, ledger.serialize());
+  }
+
+  async writeCredGraph(credGraph: CredGraph): Promise<void> {
+    const cgJson = stringify(credGraph.toJSON());
+    const outputPath = pathJoin(this._baseDirectory, ...CREDGRAPH_PATH);
+    await fs.writeFile(outputPath, cgJson);
+  }
+
+  async writeDependenciesConfig(
+    dependenciesConfig: DependenciesConfig
+  ): Promise<void> {
+    const dependenciesPath = pathJoin(
+      this._baseDirectory,
+      ...DEPENDENCIES_PATH
+    );
+    await fs.writeFile(
+      dependenciesPath,
+      stringify(dependenciesConfig, {space: 4})
+    );
+  }
+
+  //////////////////////////////
+  //
+  //  Private Functions
+  //
+  //////////////////////////////
+
+  async loadLedger(): Promise<Ledger> {
+    const ledgerPath = pathJoin(this._baseDirectory, ...LEDGER_PATH);
+    return Ledger.parse(
+      await loadFileWithDefault(ledgerPath, () => new Ledger().serialize())
+    );
+  }
+
+  async loadInstanceConfig(): Promise<InstanceConfig> {
+    const pluginsConfigPath = pathJoin(
+      this._baseDirectory,
+      ...INSTANCE_CONFIG_PATH
+    );
+    return loadJson(pluginsConfigPath, configParser);
+  }
+
+  async loadPluginGraphs(): Promise<Array<WeightedGraph>> {
+    const instanceConfig = await this.loadInstanceConfig();
+    const pluginNames = Array.from(instanceConfig.bundledPlugins.keys());
+    return await Promise.all(
+      pluginNames.map(async (name) => {
+        const outputDir = this.writePluginDirectory(name);
+        const outputPath = pathJoin(outputDir, ...GRAPHS_PATH);
+        const graphJSON = JSON.parse(await fs.readFile(outputPath));
+        return weightedGraphFromJSON(graphJSON);
+      })
+    );
+  }
+
+  async loadWeightOverrides(): Promise<WeightsT> {
+    const weightsPath = pathJoin(this._baseDirectory, ...WEIGHT_OVERRIDES_PATH);
+    return loadJsonWithDefault(weightsPath, weightsParser, emptyWeights);
+  }
+
+  async loadDependencies(): Promise<DependenciesConfig> {
+    const dependenciesPath = pathJoin(
+      this._baseDirectory,
+      ...DEPENDENCIES_PATH
+    );
+    return await loadJsonWithDefault(
+      dependenciesPath,
+      dependenciesParser,
+      () => []
+    );
+  }
+
+  async loadPluginsBudget(): Promise<Budget | null> {
+    const budgetPath = pathJoin(this._baseDirectory, ...BUDGET_PATH);
+    return loadJsonWithDefault(budgetPath, pluginBudgetParser, () => null);
+  }
+
+  writePluginDirectory(pluginId: string): string {
```

Comment:
This reads a bit confusingly to me. From the name, I’d expect it to
write a bunch of files under a plugin directory, but instead it just
creates the directory itself. Can we call it `createPluginDir`?

Reply:
accepted